### ISSUE TITLE
Firmware for ESP32-S3

### ISSUE THREAD

**Issue by smartm0use** [Blockstream/Jade#93]
Are there chances that one day Jade could be flashed on ESP32-S3 board? The one with 1,9 inch display...

![image](https://github.com/Blockstream/Jade/assets/5399265/4c1af833-4e3d-4330-9b4d-49446de56cbb)

**Comment by JamieDriver** [comment]
This could be in the works... albeit as a lower-priority background task ... ;-)  
There are some pre-requisites we need to address first - the display library/code being the main issue.  
It's not top of our list, I'll be honest - but it is something we want to support in the long run.  

Will leave this item open and will update with any progress ...

**Comment by JamieDriver** [comment]
NOTE: the button-handler library has been updated in master - which was a prerequisite for using this chip.

**Comment by JamieDriver** [comment]
Further supporting changes in master - hopefully we'll have support for this chip in the next release or two.

**Comment by smartm0use** [comment]
Great! Keep up the good work :)

**Comment by FranciscoSa** [comment]
Do we have any updates on this?

**Comment by athoszxz** [comment]
![image](https://github.com/Blockstream/Jade/assets/11262233/bdfac7f4-d624-46b1-92d8-54faeaf5e313)
I believed for a moment and failed miserably.

**Comment by JamieDriver** [comment]
No, there's a fair amount of work to get it to work on this chip ... but that work is being done when nothing more urgent takes priority.  Hopefully mid-year ...  
We'll be initially looking at the lilygo-tdisplays3 (as shown above) and the m5stack core-s3 I think.

**Comment by bobbarbosa** [comment]
I'm eagerly awaiting the release of the version for "S3". The screen is better and meets a more modern proposal. What doesn't help much is the position of the buttons or the fact that, perhaps, it isn't used vertically.
How about taking advantage of the screen and format and making it vertical?

**Comment by JamieDriver** [comment]
We have finally merged the branch adding (still early days / experimental, at this point ...) esp32s3 support.  
tdisplays3 should be supported.  
Have at it ;-)

**Comment by dark-thirty** [comment]
I'm testing new esp32s3 firmware and everything seems to be fine.

Custom board - *Waveshare ESP32-S3 DevKit N8R8, OV7670 Camera module and ILI9341 2.8" SPI TFT LCD Display*:


https://github.com/user-attachments/assets/0e86c223-45d8-4058-9c6a-fa3720f9b514

Surprisingly OV7670 camera works with ~25fps. QR code scan works OK. 

I found a little UI bug in Session/QR Scan. 
There is an outlined rectangle at the bottom of the screen, perhaps related to the position of the virtual buttons for the M5 CORE S3:

![ui](https://github.com/user-attachments/assets/81b2ff05-167c-46cd-8a94-e3a5819b501c)

ty ^_^

:ok_hand:

**Comment by JamieDriver** [comment]
Wow that's awesome!
We concentrated our initial dev/testing on the tdisplays3 and tdisplays3procamera models, so anything else may be a bit shaky atm.  Yes, someone played with the CoreS3 but I don't think it quite works yet - but as you say it does draw buttons on the touch-screen (but that ought to be controlled by `CONFIG_BOARD_TYPE_M5_CORES3` - so not 100% sure).

If it's only on the camera screen it may be an artifact from the camera or the logic that scales the image for display, etc.

Hopefully esp32-s3 will get a bit more love and polish over the next few releases.

**Comment by dark-thirty** [comment]
> Wow that's awesome!

Тhank you!

> If it's only on the camera screen it may be an artifact from the camera or the logic that scales the image for display, etc.

This is definitely not a camera artifact,  this bug only appear when i'm in an active session in jade and if i select Scan QR from the menu for example to verify wallet address.
Here is a high quality photo:

![scan_qr_bug](https://github.com/user-attachments/assets/4e34c382-c91a-42a6-9b39-ad07624bc69a)

Then if I log out and choose QR Mode, this bug goes away:

![qr_mode](https://github.com/user-attachments/assets/6c048d9a-0059-45a2-a9c0-014e26e038d2)

**Comment by JamieDriver** [comment]
ahh, I think that's the (atm empty) progress bar that would fill up if you were scanning eg a multi-frame/animated QR, like a large PSBT.

**Comment by dark-thirty** [comment]
> ie. it's not a bug, it's a feature! 😆

I see!  :joy:

Keep up the good work.

**Comment by Sandmann21s** [comment]
Do you think it would be possible to integrate this camera into a TTGO T-Display S3? This would reduce the overall size of the setup and lower the assembly costs, in addition to giving it a more professional appearance. I would be more than happy to develop a nice 3D case for it, as well as a battery module, just like I've been doing for the Jade DIY, Krux, and Seed Signer hardware wallets.



https://github.com/Blockstream/Jade/issues/93#issuecomment-2288941895

**Comment by dark-thirty** [comment]
> Do you think it would be possible to integrate this camera into a TTGO T-Display S3? 

I don't have a T-Display S3 board, but I think there are not enough free pins available.

**Comment by Sandmann21s** [comment]
The ttgo s3 is a board that has a lot of available pins and the advantage is that it already comes with a screen, which makes the final assembly more beautiful and I believe more economical, it costs around $15 to $20.
![image](https://github.com/user-attachments/assets/62b09061-a453-4d36-af47-1cc856472a61)

**Comment by greenaddress** [comment]
@smartm0use @bobbarbosa @FranciscoSa @athoszxz @dark-thirty 

master should currently work on some S3 devices. Additionally tdisplays3 and tdisplays3procamera can be flashed directly from https://blockstream.github.io/jadediyflasher/ (source code for DIY flasher is here https://github.com/Blockstream/jadediyflasher )

Do not yse the DIY flasher other than for testing/evaluation purposes. these devices in debug mode aren't really safe and the web page can't setup secure boot for you (at least for now)

Some features are still not working well across S3 devices (like USB storage) but we welcome patches.

**Comment by bubelov** [comment]
> Do not yse the DIY flasher other than for testing/evaluation purposes. these devices in debug mode aren't really safe and the web page can't setup secure boot for you (at least for now)

I saw some people on Twitter recommending that flashing tool, it would be nice to add a warning about debug mode being enabled by default and the dangers that it poses

Got an S3 Touch LCD 2 to test, works pretty well with Jade so far

**Comment by Sami32** [comment]
Tiny but handy ;-)

https://lilygo.cc/products/t-dongle-s3
